cpp/include/cudf/strings/json.hpp

```diff
@@ -26,6 +28,78 @@ namespace strings {
  * @file
  */
 
+/**
+ * @brief Settings for `get_json_object()`.
+ */
+class get_json_object_options {
+  // allow single quotes to represent strings in JSON
+  bool allow_single_quotes = false;
```

Comment:
Why should this be an option? When would we not want to support both single-quotes and double-quotes?

Reply:
Single quotes is actually not valid JSON.  This is a weird standard violation.

Reply:
Why don't throw an exception when there is a single quote? Then the users can know that they were doing wrong and will fix their data, instead of continuing to use wrong data.

Reply:
The cuIO JSON parser does not have this option. I think the caller should be required to pass correct JSON.

Reply:
That's not really an option.  This is something Spark and Hive explicitly support.

Reply:
Can you not do a `replace("\'", "\"")` first?

Reply:
Yeah, can we do one such replacement then process the result as there are only double quotes?

Reply:
Won't that break properly formed JSON string values that contain a single-quote character, e.g.: `"this is a 'test' string"`. Even if it somehow doesn't break the parsing, it will munge the string's payload value in an unrecoverable way, correct?

Reply:
I think the replacement is not just simple as `replace("\'", "\"")`, but can be done by following some pattern, such as checking the single quote and replace only if it is at the beginning of the string etc.

Reply:
The single-quote version `'this is a 'test' string'` should not exist if the flag is on as well. 
So it should be safe to just replace all single-quotes with double-quotes when you know the flag is to be used.

Reply:
> The single-quote version `'this is a 'test' string'` should not exist if the flag is on as well.
> So it should be safe to just replace all single-quotes with double-quotes when you know the flag is to be used.

That's not correct. Here's some valid JSON:

`{ "a" : "This is a 'test' "}`

If we instead had 

`{ 'a' : 'This is a 'test' '}`

Doing a replace on ' would turn it into

`{ "a" : "This is a "test" "}`

We definitely do not want this.

Regardless, even if we did a replace thing, it would be purely worse from a performance perspective.  We'd have to allocate a potentially massive intermediate buffer (the whole string column) just to do it.   That seems pretty wasteful just to avoid a bool option.

Reply:
Even with the single-quote flag, this will fail to parse: `{ 'a' : 'This is a 'test' '}` right?
So, if this cannot exist, then a replace will work.
If you control the column data  inside the Spark plugin you could do the replace in-place since its just one-byte exchanged for another.

So, maybe this should not be a libcudf function. I think we should be concerned with inconsistencies with cuIO.
@harrism @kkraus14 @jrhemstad

Reply:
> So, maybe this should not be a libcudf function. I think we should be concerned with inconsistencies with cuIO.

To reiterate, this behavior is _off_ by default.  The default behavior would match cuIO.

Reply:
For what it's worth, my intuition is that it will only be a matter of time until someone asks us to support single quotes in the cuIO JSON reader as well. Unfortunately, this is a situation that exists in the wild world of ETL and isn't something that can be easily worked around via string replacements.